### PR TITLE
Creating admin panel for advice 1784

### PR DESCRIPTION
**Pull request by Markiyan000**

### CODE REVIEW

**Review by Yura09 — APPROVED** [review]

**Review by forestik — APPROVED** [review]

File: core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import springfox.documentation.annotations.ApiIgnore;
+
+import javax.validation.Valid;
+import java.util.List;
+
+import static greencity.dto.genericresponse.GenericResponseDto.buildGenericResponseDto;
+
+@Controller
```
Comment (vterlyha) [inline core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java:29]: Mark it as RestController and you won't need to add ResponseBody annotation to methods

File: core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    @GetMapping
+    public String findAllAdvices(@RequestParam(required = false) String query,
+        Model model, @ApiIgnore Pageable pageable) {
+        PageableDto<AdviceVO> allAdvices = query == null || query.isEmpty()
+            ? adviceService.getAllAdvices(pageable)
+            : adviceService.searchBy(pageable, query);
+        model.addAttribute("pageable", allAdvices);
+        model.addAttribute("languages", languageService.getAllLanguages());
+
+        return "core/management_advices";
```
Comment (vterlyha) [inline core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java:60]: What is it ?

File: core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping
+    public String findAllAdvices(@RequestParam(required = false) String query,
+        Model model, @ApiIgnore Pageable pageable) {
+        PageableDto<AdviceVO> allAdvices = query == null || query.isEmpty()
+            ? adviceService.getAllAdvices(pageable)
+            : adviceService.searchBy(pageable, query);
```
Comment (vterlyha) [inline core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java:56]: Move business logic to Service method

File: core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    public String filterAdvices(Model model, @ApiIgnore Pageable pageable,
+        AdviceViewDto adviceViewDto) {
+        PageableDto<AdviceVO> filteredAdvices =
+            adviceService.getFilteredAdvices(pageable, adviceViewDto);
+        model.addAttribute("pageable", filteredAdvices);
+        model.addAttribute("fields", adviceViewDto);
+
+        return "core/management_advices";
```
Comment (vterlyha) [inline core/src/main/java/greencity/webcontroller/ManagementAdvicesController.java:187]: Same here ?

File: dao/src/main/java/greencity/repository/AdviceRepo.java [hunk 5]
```diff
@@ -2,12 +2,48 @@
...
+     * @author Markiyan Derevetskyi
+     */
+    @Query(value = "select distinct a from Advice a join fetch a.translations as at "
+        + "where concat(a.id, '') like lower(concat(:query, '')) "
+        + "or concat(a.habit.id, '') like lower(concat(:query, '')) "
+        + "or concat(at.id, '') like lower(concat(:query, '')) "
+        + "or lower(at.language.code) like lower(concat('%', :query, '%')) "
+        + "or lower(at.content) like lower(concat('%', :query, '%'))",
+        countQuery = "select count(a) from Advice a")
+    Page<Advice> searchBy(Pageable pageable, String query);
```
Comment (vterlyha) [inline dao/src/main/java/greencity/repository/AdviceRepo.java:45]: Bad method naming and bad parameter naming

**Review by IvanBegar — APPROVED** [review]

**Pull request merged**